platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheUnsupportedTypesIntegrationTest.groovy

```diff
@@ -394,4 +396,81 @@ class ConfigurationCacheUnsupportedTypesIntegrationTest extends AbstractConfigur
+        file("buildSrc/src/main/java/JavaRecord.java") << """
+            public record JavaRecord(${baseType.name} value, String filler) {}
+        """
+        file("buildSrc/build.gradle.kts") << """
+            plugins {
+                `java-library`
+            }
+        """
+        buildFile << """
+            class SomeBean {
+                JavaRecord value
+            }
+
+            class SomeTask extends DefaultTask {
+                private final SomeBean bean = new SomeBean()
+                private final JavaRecord value
+
+                SomeTask() {
+                    ${creator}
+                    value = new JavaRecord(${reference}, "some data")
+                    bean.value = new JavaRecord(${reference}, "some other data")
+                }
+
+                @TaskAction
+                void run() {
+                    println "this.reference = " + value
+                    println "bean.reference = " + bean.value
+                }
+            }
+
+            task broken(type: SomeTask)
+        """
+
+        when:
+        configurationCacheRunLenient "broken"
+
+        then:
+        problems.assertResultHasProblems(result) {
+            withTotalProblemsCount(6)
+            withUniqueProblems(
+                "Task `:broken` of type `SomeTask`: cannot deserialize object of type '${baseType.name}' as these are not supported with the configuration cache.",
+                "Task `:broken` of type `SomeTask`: cannot serialize object of type '${concreteType.name}', a subtype of '${baseType.name}', as these are not supported with the configuration cache.",
+                "Task `:broken` of type `SomeTask`: value '$deserializedValue' is not assignable to '${baseType.name}'"
+            )
+            withProblemsWithStackTraceCount(0)
+        }
+
+        when:
+        configurationCacheRunLenient "broken"
+
+        and:
+        outputContains("this.reference = JavaRecord[value=null, filler=some data]")
+        outputContains("bean.reference = JavaRecord[value=null, filler=some other data]")
+
+        then:
+        problems.assertResultHasProblems(result) {
+            withTotalProblemsCount(4)
+            withUniqueProblems(
+                "Task `:broken` of type `SomeTask`: cannot deserialize object of type '${baseType.name}' as these are not supported with the configuration cache.",
+                "Task `:broken` of type `SomeTask`: value '$deserializedValue' is not assignable to '${baseType.name}'"
+            )
+            withProblemsWithStackTraceCount(0)
+        }
+
+        and:
+        outputContains("this.reference = JavaRecord[value=null, filler=some data]")
+        outputContains("bean.reference = JavaRecord[value=null, filler=some other data]")
+
+        where:
+        concreteType                     | baseType           | creator                                     | reference                                            | deserializedValue
```

Comment:
❌  I'd add some unsupported live JVM type to this test too, like a `Thread` or a `FileInputStream`.